compiler/luci-interpreter/src/kernels/FloorMod.test.cpp

```diff
@@ -0,0 +1,358 @@
+
+  Shape input2_shape = input1_shape;
+  std::vector<int16_t> input2_data{2, 2, -3, -4};
+
+  std::vector<int32_t> ref_output_shape{1, 2, 2, 1};
+  std::vector<int16_t> ref_output_data{0, 1, -2, -1};
+
+  Tensor input1_tensor =
+    makeInputTensor<DataType::S16>(input1_shape, input1_data, _memory_manager.get());
+  Tensor input2_tensor =
+    makeInputTensor<DataType::S16>(input2_shape, input2_data, _memory_manager.get());
+  Tensor output_tensor = makeOutputTensor(DataType::S16);
+
+  FloorMod kernel(&input1_tensor, &input2_tensor, &output_tensor);
+  kernel.configure();
+  _memory_manager->allocate_memory(output_tensor);
+  kernel.execute();
+
+  EXPECT_THAT(extractTensorData<int16_t>(output_tensor),
+              ::testing::ElementsAreArray(ref_output_data));
+  EXPECT_THAT(extractTensorShape(output_tensor), ::testing::ElementsAreArray(ref_output_shape));
+}
+
+TEST_F(FloorModTest, BroadcastFloorModInt16)
+{
+  Shape input1_shape{1, 2, 2, 1};
+  std::vector<int16_t> input1_data{10, -9, -11, 7};
+
+  Shape input2_shape{1};
+  std::vector<int16_t> input2_data{-3};
+
+  std::vector<int32_t> ref_output_shape{1, 2, 2, 1};
+  std::vector<int16_t> ref_output_data{-2, 0, -2, -2};
+
+  Tensor input1_tensor =
+    makeInputTensor<DataType::S16>(input1_shape, input1_data, _memory_manager.get());
+  Tensor input2_tensor =
+    makeInputTensor<DataType::S16>(input2_shape, input2_data, _memory_manager.get());
+  Tensor output_tensor = makeOutputTensor(DataType::S16);
+
+  FloorMod kernel(&input1_tensor, &input2_tensor, &output_tensor);
+  kernel.configure();
+  _memory_manager->allocate_memory(output_tensor);
+  kernel.execute();
+
+  EXPECT_THAT(extractTensorData<int16_t>(output_tensor),
+              ::testing::ElementsAreArray(ref_output_data));
+  EXPECT_THAT(extractTensorShape(output_tensor), ::testing::ElementsAreArray(ref_output_shape));
+}
+
+TEST_F(FloorModTest, DivByZero_NEG)
+{
+  Shape shape{3};
+  std::vector<int32_t> input1_data{1, 0, -1};
+  std::vector<int32_t> input2_data{0, 0, 0};
+
+  Tensor input1_tensor = makeInputTensor<DataType::S32>(shape, input1_data, _memory_manager.get());
+  Tensor input2_tensor = makeInputTensor<DataType::S32>(shape, input2_data, _memory_manager.get());
+  Tensor output_tensor = makeOutputTensor(DataType::S32);
+
+  FloorMod kernel(&input1_tensor, &input2_tensor, &output_tensor);
+  kernel.configure();
+  _memory_manager->allocate_memory(output_tensor);
+
+  EXPECT_ANY_THROW(kernel.execute());
+}
+
+TEST_F(FloorModTest, Input_Output_Type_Mismatch_NEG)
+{
+  Tensor input1_tensor = makeInputTensor<DataType::FLOAT32>({1}, {1.f}, _memory_manager.get());
+  Tensor input2_tensor = makeInputTensor<DataType::FLOAT32>({1}, {1.f}, _memory_manager.get());
+  Tensor output_tensor = makeOutputTensor(DataType::S8);
+
+  FloorMod kernel(&input1_tensor, &input2_tensor, &output_tensor);
+  EXPECT_ANY_THROW(kernel.configure());
+}
+
+TEST_F(FloorModTest, Input_Type_Mismatch_NEG)
```

Comment:
Can you please add num_of(NEG test case) >= num_of(POS test case) ?

Reply:
Yes, I will.

Reply:
I added NEG tests.